### ISSUE TITLE
Allow setting custom folder icons

### ISSUE THREAD

**Issue by chuchanpr** [files-community/Files#1015]
**Is your feature request related to a problem? Please describe.**
Allow setting customer folder icons

**Describe the solution you'd like**
Current folder options does not allow setting customer icons

**Describe alternatives you've considered**
Set customer icon using windows explorer. but same settings do not apply

**Comment by mdtauk** [comment]
How about allowing Icon Packs to replace common icons in one go from settings - as well as changing individual icons

**Comment by yaichenbaum** [comment]
@mdtauk Do you have in a mind a dialog with a preset of icons users can choose from?
![image](https://user-images.githubusercontent.com/39923744/84456745-cada5200-ac2e-11ea-9da1-db8532e692fd.png)

**Comment by mdtauk** [comment]
> @mdtauk Do you have in a mind a dialog with a preset of icons users can choose from?
> ![image](https://user-images.githubusercontent.com/39923744/84456745-cada5200-ac2e-11ea-9da1-db8532e692fd.png)

That is of course the way Win32 does it. I probably think it would be more of a Change Icon button, which opens a file picker dialog, allowing you to pick an *.ico /.png /.svg  etc file.

As for an icon pack, I guess you would need some kind of zip file containing the image files and possibly some kind of JSON file assigning or naming the folder and file types, each icon is for.  This would be done kind of format you decide on.

Ubuntu and Android supports custom icon packs which when loaded by a launcher or shell tweaks tool, will change a bunch of icons in one go.

**Comment by IdmFoundInHim** [comment]
Besides a new dialog, the existing mechanism from Windows File Explorer should be supported or (at least) importable. Folders that already have a custom icon will have the following in that folder's `desktop.ini`:
```
[.ShellClassInfo]
IconResource=C:\file\path\here\icon.ico,0
```
(The 0 would be a different number if the targeted file contains multiple icons.)

If Files is not going to support the `desktop.ini` setting but supports custom folder icons in some other way, there should be a GUI way to transfer the existing settings to the supported format.

**Comment by gave92** [comment]
Curious: I remember seeing some work by @duke7553 for this, what happened to it?

**Comment by yaichenbaum** [comment]
This issue should be resolved with #6639.
